dimensiondata/src/main/java/org/jclouds/dimensiondata/cloudcontrol/compute/strategy/GetOrCreateNetworkDomainThenCreateNodes.java

```diff
@@ -0,0 +1,175 @@
+import org.jclouds.compute.strategy.CustomizeNodeAndAddToGoodMapOrPutExceptionIntoBadMap;
+import org.jclouds.compute.strategy.ListNodesStrategy;
+import org.jclouds.compute.strategy.impl.CreateNodesWithGroupEncodedIntoNameThenAddToSet;
+import org.jclouds.dimensiondata.cloudcontrol.DimensionDataCloudControlApi;
+import org.jclouds.dimensiondata.cloudcontrol.compute.options.DimensionDataCloudControlTemplateOptions;
+import org.jclouds.dimensiondata.cloudcontrol.domain.NetworkDomain;
+import org.jclouds.dimensiondata.cloudcontrol.domain.State;
+import org.jclouds.dimensiondata.cloudcontrol.domain.Vlan;
+import org.jclouds.dimensiondata.cloudcontrol.predicates.NetworkDomainState;
+import org.jclouds.dimensiondata.cloudcontrol.predicates.VlanState;
+
+import javax.inject.Singleton;
+import java.util.Map;
+import java.util.Set;
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+import static java.lang.String.format;
+import static org.jclouds.util.Predicates2.retry;
+
+@Singleton
+public class GetOrCreateNetworkDomainThenCreateNodes extends CreateNodesWithGroupEncodedIntoNameThenAddToSet {
+
+   private final DimensionDataCloudControlApi api;
+   private final ComputeServiceConstants.Timeouts timeouts;
+
+   public static final String DEFAULT_NETWORK_DOMAIN_NAME = "JCLOUDS_NETWORK_DOMAIN";
+   public static final String DEFAULT_VLAN_NAME = "JCLOUDS_VLAN";
+   public static final String DEFAULT_PRIVATE_IPV4_BASE_ADDRESS = "10.0.0.0";
+   public static final Integer DEFAULT_PRIVATE_IPV4_PREFIX_SIZE = 24;
+
+   @Inject
+   protected GetOrCreateNetworkDomainThenCreateNodes(final CreateNodeWithGroupEncodedIntoName addNodeWithGroupStrategy,
+         final ListNodesStrategy listNodesStrategy, final GroupNamingConvention.Factory namingConvention,
+         final ListeningExecutorService userExecutor,
+         final CustomizeNodeAndAddToGoodMapOrPutExceptionIntoBadMap.Factory customizeNodeAndAddToGoodMapOrPutExceptionIntoBadMapFactory,
+         final DimensionDataCloudControlApi api, final ComputeServiceConstants.Timeouts timeouts) {
+      super(addNodeWithGroupStrategy, listNodesStrategy, namingConvention, userExecutor,
+            customizeNodeAndAddToGoodMapOrPutExceptionIntoBadMapFactory);
+      this.api = api;
+      this.timeouts = timeouts;
+   }
+
+   @Override
+   public Map<?, ListenableFuture<Void>> execute(final String group, final int count, final Template template,
+         final Set<NodeMetadata> goodNodes, final Map<NodeMetadata, Exception> badNodes,
+         final Multimap<NodeMetadata, CustomizationResponse> customizationResponses) {
+
+      final DimensionDataCloudControlTemplateOptions templateOptions = template.getOptions()
+            .as(DimensionDataCloudControlTemplateOptions.class);
+
+      String networkDomainName = firstNonNull(templateOptions.getNetworkDomainName(), DEFAULT_NETWORK_DOMAIN_NAME);
+      String vlanName = firstNonNull(templateOptions.getVlanName(), DEFAULT_VLAN_NAME);
```

Comment:
The jclouds portable template options have the `networks` attribute (names, ids, or whatever makes sense for the provider) so users can configure to which networks the nodes will be connected to. Instead of providing two different options to configure that, we should try to accommodate the current behavior to using that `networks` option in a way that is understandable and makes sense to users.